plugins/aws-catalog-backend/src/BaseEntityProvider.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright SeatGeek
+ * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
+ */
+import { TaskRunner } from '@backstage/backend-tasks';
+import {
+  ANNOTATION_LOCATION,
+  ANNOTATION_ORIGIN_LOCATION,
+  ResourceEntity,
+} from '@backstage/catalog-model';
+import {
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-node';
+import { arn } from '@seatgeek/backstage-plugin-aws-catalog-common';
+import { merge } from 'lodash';
+import * as uuid from 'uuid';
+import { Logger } from 'winston';
+
+export type InstanceTransformer<T> = (instance: T) => Promise<ResourceEntity>;
+export type InstanceFilter<T> = (instance: T) => boolean;
+
+export abstract class BaseEntityProvider<T> implements EntityProvider {
+  protected readonly logger: Logger;
+  private readonly scheduleFn: () => Promise<void>;
+  private connection?: EntityProviderConnection;
+  protected id?: string;
```

Comment:
this can be `readonly`, no?

Reply:
Yep, and private too.  Fixed via c6c252d